x/mongo/driver/topology/fsm.go

```diff
@@ -20,13 +21,17 @@ var minSupportedMongoDBVersion = "2.6"
 
 type fsm struct {
 	description.Topology
-	SetName       string
-	maxElectionID primitive.ObjectID
-	maxSetVersion uint32
+	SetName          string
+	maxElectionID    primitive.ObjectID
+	maxSetVersion    uint32
+	compatible       atomic.Value
```

Comment:
Just curious, why does "compatible" need to be atomic?

Reply:
When a new server gets added to a topology, it sets up a background goroutine that can data race with the tests when the tests are checking compatibility, and storing it as an atomic value prevents that.